src/app/core/identity-provider/co-browse.identity-provider.ts

```diff
@@ -0,0 +1,120 @@
+import { HttpEvent, HttpHandler, HttpRequest } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, Router } from '@angular/router';
+import { Store, select } from '@ngrx/store';
+import { Observable, noop, of, race, throwError } from 'rxjs';
+import { catchError, concatMap, delay, filter, first, map, switchMap, take, withLatestFrom } from 'rxjs/operators';
+
+import { AccountFacade } from 'ish-core/facades/account.facade';
+import { AppFacade } from 'ish-core/facades/app.facade';
+import { CheckoutFacade } from 'ish-core/facades/checkout.facade';
+import { IdentityProvider, TriggerReturnType } from 'ish-core/identity-provider/identity-provider.interface';
+import { selectQueryParam } from 'ish-core/store/core/router';
+import { ApiTokenService } from 'ish-core/utils/api-token/api-token.service';
+import { whenTruthy } from 'ish-core/utils/operators';
+
+@Injectable({ providedIn: 'root' })
+export class CoBrowseIdentityProvider implements IdentityProvider {
+  constructor(
+    protected router: Router,
+    protected store: Store,
+    protected apiTokenService: ApiTokenService,
+    private appFacade: AppFacade,
+    private accountFacade: AccountFacade,
+    private checkoutFacade: CheckoutFacade
+  ) {}
+
+  getCapabilities() {
+    return {
+      editPassword: false,
+      editEmail: false,
+      editProfile: false,
+    };
+  }
+
+  init() {
+    this.apiTokenService.cookieVanishes$
+      .pipe(withLatestFrom(this.apiTokenService.apiToken$))
+      .subscribe(([type, apiToken]) => {
+        if (!apiToken) {
+          this.accountFacade.fetchAnonymousToken();
+        }
+        if (type === 'user') {
+          this.accountFacade.logoutUser({ revokeApiToken: false });
+        }
+      });
+
+    // OAuth Service should be configured before apiToken information are restored and the refresh token mechanism is setup
+    this.apiTokenService.restore$(['user', 'order']).subscribe(noop);
```

Comment:
We have implemented something similar for our client and we have observed that token is not encoded by ICM URL, so we need to reinsert '+'
``` accessToken = decodeURIComponent(accessToken?.replace(/\s/g, '+'));```